suite/tests/client-interface/syscall-mod.c

```diff
@@ -52,20 +52,26 @@ main()
     asm("movz x8, " STRINGIFY(SYS_getpid) ";"
                                           "svc 0;"
                                           "mov %w0, w0"
-        : "=r"(pid));
+        : "=r"(pid)
+        :
+        : "cc", "w0");
 #elif defined(X64)
     /* we don't want vsyscall since we rely on mov immed, eax being in same bb.
      * plus, libc getpid might cache the pid value.
      */
     asm("mov $" STRINGIFY(SYS_getpid) ", %%eax;"
                                       "syscall;"
                                       "mov %%eax, %0"
-        : "=m"(pid));
+        : "=m"(pid)
+        :
+        : "cc", "rcx", "r11", "rax");
```

Comment:
I think it might be macOS specifics, but it looks like the flags are modded here. Have a look on macOS libsystem_kernel code: https://github.com/apple/darwin-xnu/blob/xnu-7195.121.3/libsyscall/custom/SYS.h#L134-L142. This code uses the flags to determine whether the syscall succeeded.

Reply:
Ah yes that is right: for BSD syscalls the return path out of the kernel sets the flags, and DR has to look at CF to check for success: https://github.com/DynamoRIO/dynamorio/blob/master/core/unix/os.c#L5173